click/utils.py

```diff
@@ -211,11 +211,17 @@ def echo(message=None, file=None, nl=True, err=False, color=None):
     :param color: controls if the terminal supports ANSI colors or not.  The
                   default is autodetection.
     """
-    if file is None:
+    if is_running_jupyter():
         if err:
-            file = _default_text_stderr()
+            file = sys.stderr
         else:
-            file = _default_text_stdout()
+            file = sys.stdout
+    else:
```

Comment:
There's this interesting little construct called "else if" …

Reply:
there is this other interesting little construct called “usefulness”. Your previous comment could have some of it. It could also have some of “nice” or at least “polite” but that might be asking too much.

Stupid comments aside; the reason why it’s written like this is readability. The split between the cases is much clearer than conflating the condition and writing 4 or 5 different cases. Saying that, happy to rewrite if the maintainers prefer the other way.